ext/session/session.c

```diff
@@ -2667,7 +2672,8 @@ PHP_FUNCTION(session_start)
 					case IS_FALSE:
 					case IS_LONG:
 						if (zend_string_equals_literal(str_idx, "read_and_close")) {
-							read_and_close = zval_get_long(value);
+							zend_long tmp = zval_get_long_ex(value, true);
```

Comment:
let me know if you think zval_try_get_long is more appropriate, so we forgo "true"/"false" for example.

Reply:
I think `zval_get_long()` here is fine, except for the string case, probably best to handle this manually with one of the numeric string APIs.